api/client.go

```diff
@@ -336,12 +337,23 @@ func (c *Client) Clone() (*Client, error) {
 // configured for this client. This is an advanced method and generally
 // doesn't need to be called externally.
 func (c *Client) NewRequest(method, requestPath string) *Request {
+	// if SRV records exist (see https://tools.ietf.org/html/draft-andrews-http-srv-02), lookup the SRV
```

Comment:
Apologies - I hadn't intended to merge this commit.  I think it's a useful feature though, I'll disentangle it from the PR.